src/osp/scientific/kepler.cpp

```diff
@@ -133,63 +143,63 @@ static double mean_motion(double const mu, double const a)
 
 static double get_orbiting_radius(double const ecc, double const semiMajorAxis, double const trueAnomaly)
```

Comment:
all of these math functions look like they are `noexcept` right?

Reply:
i am a noob at c++ but in (76-77)  of kerpler.cpp  double df_E = 1.0 - eccentricity * cos(E); // if eccentricity happens to be 1,  a div by 0   , since c++ do-while loops   execute at least once,  idk if that causes issues in c++ but every language i actually have skills in throws a exception

Reply:
my understanding is that floating point exceptions are not the same as C++ exceptions. I think if it does a divide by zero, the program will simply crash

regardless, if crashing isn't the right answer, then what *should* the function do?

Reply:
More accurately,  on posix platforms it'll receive a signal that interrupts program flow and calls a signal handler (which we have no handler registered, so it'll default to crashing)

on windows it'll likely result in a structured exception handler exception being thrown.

Regardless, throwing a c++ exception isnt the right answer, we need to have a solid determination of what we want the program to do if div by zero happens in this function.

Crashing is one choice, or we can return some kind of error code, but we want to document what the behaivor is going to be and then stick to it.

Reply:
A floating point math error isn't necessarily an unrecoverable error for the end application, so I don't know if crashing is a good default choice choice for handling them.

I'm not well versed in c++ error handling though, so I don't know what the right way to handle floating point errors would be.

Reply:
My recommendation is to add `noexcept` unless someone provides a signal-handler -> exception translator, which very well might happen.

Until then, `noexcept` can result in better code-gen in many situations.